gp-includes/formats/format-android.php

```diff
@@ -44,108 +100,485 @@ public function print_exported_file( $project, $locale, $translation_set, $entri
+		$contents = str_ireplace( '</xliff:g>', '--xlifftag--/xliff:g>', $contents );
+
+		// Parse the file contents.
+		$data = simplexml_load_string( $contents, null, LIBXML_NOCDATA );
+
+		// Reset the error display to it's original setting.
 		libxml_use_internal_errors( $errors );
 
+		// Check to see if the XML parsing was successful.
 		if ( ! is_object( $data ) )
 			return false;
 
 		$entries = new Translations;
 
+		// Loop through all of the single strings we found in the XML file.
 		foreach ( $data->string as $string ) {
+			// If the string is marked as non-translatable, skip it.
 			if ( isset( $string['translatable'] ) && 'false' == $string['translatable'] ) {
 				continue;
 			}
 
-			$entry = new Translation_Entry();
-			$entry->context = (string)$string['name'];
-			$entry->singular = $this->unescape( (string)$string[0] );
-			$entry->translations = array();
-
-			if ( isset( $string['comment'] ) && $string['comment'] ) {
-				$entry->extracted_comments = $string['comment'];
-			}
+			// Generate the entry to add.
+			$entry = $this->generate_entry( $string, (string) $string['name'] );
 
+			// Add the entry to the results.
 			$entries->add_entry( $entry );
 		}
 
-		foreach ( $data->{'string-array'} as $string_array )
-		{
+		// Loop through all of the multiple strings we found in the XML file.
+		foreach ( $data->{'string-array'} as $string_array ) {
 			if ( isset( $string_array['translatable'] ) && 'false' == $string_array['translatable'] ) {
 				continue;
 			}
 
 			$array_name = (string) $string_array['name'];
 			$item_index = 0;
 
-			foreach ( $string_array->item as $item )
-			{
-				$entry               = new Translation_Entry();
-				$entry->context      = $array_name . "[$item_index]";
-				$entry->singular     = $this->unescape( $item[0] );
-				$entry->translations = array();
+			foreach ( $string_array->item as $string ) {
+				// Generate the entry to add.
+				$entry = $this->generate_entry( $string, $array_name . "[$item_index]" );
 
+				// Add the entry to the results.
 				$entries->add_entry( $entry );
 
+				// Increment our index for the next entry.
 				$item_index++;
 			}
 		}
 
+		// Loop through all of the plural entries we found in the XML file.
+		foreach ( $data->plurals as $plural ) {
+			if ( isset( $plural['translatable'] ) && 'false' == $plural['translatable'] ) {
+				continue;
+			}
+
+			$array_name = (string) $plural['name'];
+			$item_index = 0;
+
+			foreach ( $plural->item as $string ) {
+				$strings[] = $string;
+			}
+
+			// Generate the entry to add.
+			$entry = $this->generate_plural_entry( $strings, $array_name );
+
+			// Add the entry to the results.
+			$entries->add_entry( $entry );
+		}
+
 		return $entries;
 	}
 
+	/**
+	 * Generates a translation entry object to be added to the results for the "read_originals_from_file()" function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param obj    $string  The string entry object to use.
```

Comment:
It seems like `$string` is an array.

Generally, if the type is an object than `object` should be used.